invoices/invoiceregistry_test.go

```diff
@@ -240,6 +241,69 @@ func TestSettleInvoice(t *testing.T) {
 	}
 }
 
+// Tests that invoice is rejected after expiration.
+func TestInvoiceExpiry(t *testing.T) {
+	registry, cleanup := newTestContext(t)
+	defer cleanup()
+
+	allSubscriptions := registry.SubscribeNotifications(0, 0)
+	defer allSubscriptions.Cancel()
+
+	const testExpiry = time.Second
+
+	expiringInvoice := &channeldb.Invoice{
+		Terms: channeldb.ContractTerm{
+			PaymentPreimage: preimage,
+			Value:           lnwire.MilliSatoshi(100000),
+		},
+		CreationDate: time.Now(),
```

Comment:
Better to use a fixed time for testing, to make it deterministic.

Reply:
Sidenote: There are two tests now. One tests AddInvoice with an expired timestamp (one day in the past). The other checking expration with a valid timestamp.

re fixed time: I'm a bit unsure how I can make this completely fixed as expiration checking uses time.Now() as a reference point. Is there maybe a way to mock time in tests?

Reply:
We solved this problem elsewhere by adding a field `now func() time.Time` to `InvoiceRegistry`. That way the test can control the time.

Reply:
done